deltacat/aws/s3u.py

```diff
@@ -121,6 +118,31 @@ def _get_write_path_for_block(
             self.block_refs.append(block)
         return write_path
 
+    def __call__(
+        self,
+        base_path: str,
+        *,
+        filesystem: Optional[pa.filesystem.FileSystem] = None,
+        dataset_uuid: Optional[str] = None,
+        block: Optional[ObjectRef[Block]] = None,
+        block_index: Optional[int] = None,
+        file_format: Optional[str] = None,
+    ) -> str:
+        """
+        TODO: BlockWritePathProvider is deprecated as of Ray version 2.20.0. Please use FilenameProvider.
+        See: https://docs.ray.io/en/master/data/api/doc/ray.data.datasource.FilenameProvider.html
+
+        Hence, this class only works with Ray version 2.3.0 or lower when used in Ray Dataset.
+        """
```

Comment:
nits: 
Maybe the docstring could be more explicit about what we're doing here (implementing __call__ as a temporary workaround for the fact that it's no longer implemented for the deprecated superclass BlockPathProvider).

Probably we should also create a GitHub issue for the TODO and link it in the comment.

Reply:
done.

Reply:
+1, did not understand what was happening until I read Dmitri's comment